core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -19,14 +23,14 @@ final case class NonEmptyVector[A] private (toVector: Vector[A]) {
 
   /** Updates the element at the index, if it exists */
   def updated(i: Int, a: A): Option[NonEmptyVector[A]] =
-    if (toVector.isDefinedAt(i)) Some(NonEmptyVector(toVector.updated(i, a))) else None
+    if (toVector.isDefinedAt(i)) Some(new NonEmptyVector(toVector.updated(i, a))) else None
```

Comment:
Yea we could do that. For reference `Vector` inherits `isDefinedAt` from `GenSeqLike` - https://github.com/scala/scala/blob/v2.11.8/src/library/scala/collection/GenSeqLike.scala#L64-L72

Definition is pretty much the same, with the exception that a local implementation can be private and inlined.